pkg/kubectl/metricsutil/metrics_printer.go

```diff
@@ -88,69 +93,91 @@ func (printer *TopCmdPrinter) PrintPodMetrics(metrics []metricsapi.PodMetrics, p
 		printValue(w, PodColumn)
 	}
 	printColumnNames(w, PodColumns)
-	for _, m := range metrics {
-		err := printSinglePodMetrics(w, &m, printContainers, withNamespace)
-		if err != nil {
-			return err
-		}
-	}
-	return nil
-}
 
-func printColumnNames(out io.Writer, names []string) {
-	for _, name := range names {
-		printValue(out, name)
+	allContainers, pods, err := addContainerMetrics(metrics, printContainers)
+	if err != nil {
+		return err
 	}
-	fmt.Fprint(out, "\n")
-}
 
-func printSinglePodMetrics(out io.Writer, m *metricsapi.PodMetrics, printContainersOnly bool, withNamespace bool) error {
-	containers := make(map[string]api.ResourceList)
-	podMetrics := make(api.ResourceList)
-	for _, res := range MeasuredResources {
-		podMetrics[res], _ = resource.ParseQuantity("0")
+	if !printContainers {
+		sort.Sort(podMetricsSort{metrics, pods, sorting})
+		for idx, m := range metrics {
```

Comment:
It feels like this should maybe go in a printAllPods function to match line 112.